org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/Document.java

```diff
@@ -38,16 +39,38 @@ public Document(Integer version, String contents, boolean printSourceOnError) {
 		this.contents = contents;
 		this.printSourceOnError = printSourceOnError;
 	}
+	
+	public record Range(int startOffset, int endOffset) {
```

Comment:
we also already have

- ITextRegion (well offset + length)

but we only partially use it in xtext.ide but e.g. Formatting Service does already use it